app/services/hyrax/characterization/file_set_description.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Hyrax
+  module Characterization
+    ##
+    # @api public
+    class FileSetDescription
+      include Hydra::Works::MimeTypes
+
+      ##
+      # @!attribute [rw] file_set
+      #   @return [Hyrax::FileSet]
+      attr_accessor :file_set
+
+      delegate :mime_type, to: :primary_file
+
+      ##
+      # @param [Hyrax::FileSet] file_set
+      # @param [Symbol] primary_file  a symbol mapping to the file_set member
+      #   used for characterization
+      def initialize(file_set:, primary_file: :original_file)
+        self.file_set = file_set
+        @primary_file = primary_file
+      end
+
+      ##
+      # @api public
+      # @return [Hyrax::FileMetadata] the member file to use for characterization
+      def primary_file
```

Comment:
Something of this sits a bit odd. Setting the `primary_file` based on a symbol, but returning an `Hyrax::FileMetadata` object. It feels odd to have an assigned instance_variable that is different from the returned value from the method named the same as the instance_variable.

Do we want / need memoization for the primary_file (as FileMetadata)?

Reply:
i think my expectation was that callers would want to use a symbol since this is how the existing `characterization_proxy` class attribute is handled.

do you have suggestions about alternate approaches?

Reply:
If I had my druthers, I might look at renaming the initialize parameter to `primary_file_type`. Though in typing that I'm not overly satisfied. However, to keep backwards API compatability, I think what you have is reasonable. The file/class is not that difficult to follow, as there is little going on.

Reply:
:+1: i definitely get the awkwardness, but don't have a better solution at hand.

Reply:
I'm good with leaving as is. The compact nature of the class keeps this cognitive load to what I think is reasonable.